### PR TITLE
LPS-202928 & LPS-202929 Update the translation when the event from localized input is listened

### PR DESCRIPTION
**Pull request by veroglez**
This PR depends on https://github.com/liferay-echo/liferay-portal/pull/14573. Just to review from ad2618d8feb019abd7b6620bd37c9067eb7406d5

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by veroglez** [comment]
ci:test:sf

**Review by p2kmgcl — CHANGES_REQUESTED** [review]

File: modules/apps/layout/layout-js-components-web/src/main/resources/META-INF/resources/js/components/translation_selector/TranslationSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+import {Option, Picker} from '@clayui/core';
+import ClayIcon from '@clayui/icon';
+import ClayLayout from '@clayui/layout';
+import {useId} from 'frontend-js-components-web';
+import React, {useState} from 'react';
+
+const TriggerLabel = React.forwardRef(
+	(
+		{selectedItem, ...otherProps}: {selectedItem: Language},
```
[the remark this answers is not in the document]
Comment (veroglez) [reply modules/apps/layout/layout-js-components-web/src/main/resources/META-INF/resources/js/components/translation_selector/TranslationSelector.tsx]: I remove this component in this commit f56dcdaa809cf1ca824d59a95d9c51d3e666c64f, it's just that it was in a PR that had already passed :sweat_smile:

File: modules/apps/layout/layout-js-components-web/src/main/resources/META-INF/resources/js/components/translation_selector/TranslationSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+);
+export interface Language {
+	icon: string;
+	id: Liferay.Language.Locale;
+	label: Liferay.Language.Locale;
+	translations: number;
+}
+
+export interface Translations {
+	[key: string]: Record<Liferay.Language.Locale, string>;
```
[the remark this answers is not in the document]
Comment (veroglez) [reply modules/apps/layout/layout-js-components-web/src/main/resources/META-INF/resources/js/components/translation_selector/TranslationSelector.tsx]: Same here, the component is `TranslationSelector` :sweat:

File: modules/apps/layout/layout-js-components-web/src/main/resources/META-INF/resources/js/components/translation_selector/TranslationSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+export default function TranslationSelector({
+	defaultLanguageId,
+	languages,
+	onSelectedLanguageChange,
+	selectedLanguageId,
+	translations,
+}: Props) {
+	const selectorId = useId();
+
+	const getSelectedLanguage = (id: React.Key) =>
```
[the remark this answers is not in the document]
Comment (veroglez) [reply modules/apps/layout/layout-js-components-web/src/main/resources/META-INF/resources/js/components/translation_selector/TranslationSelector.tsx]: Same here

File: modules/apps/layout/layout-js-components-web/src/main/resources/META-INF/resources/js/components/translation_selector/TranslationSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+		languages.find((language) => language.id === id)!;
+
+	const [selectedLanguage, setSelectedLanguage] = useState<Language>(
+		getSelectedLanguage(selectedLanguageId)!
+	);
+
+	const items = translations
```
[the remark this answers is not in the document]
Comment (veroglez) [reply modules/apps/layout/layout-js-components-web/src/main/resources/META-INF/resources/js/components/translation_selector/TranslationSelector.tsx]: Same here

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/translation_manager/TranslationManager.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+		fieldToTranslation(fields)
+	);
+
+	const getTranslations = useCallback(
+		() =>
+			Object.keys(fields).map((fieldName) => {
+				const languages = Array.from(
+					document.querySelectorAll<HTMLInputElement>(
+						`[type="hidden"][id*="${portletNamespace}${fieldName}_"]`
+					)
```
[the remark this answers is not in the document]
Comment (veroglez) [reply modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/translation_manager/TranslationManager.tsx:44]: Added!

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/translation_manager/TranslationManager.tsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+				return {
+					...acc,
+					...(translatedItems && {[locale.id]: translatedItems}),
+				};
+			}, {}),
+		[translations, locales]
+	);
+
+	useEffect(() => {
+		Liferay.on('inputLocalized:updateTranslationStatus', () =>
```
[the remark this answers is not in the document]
Comment (veroglez) [reply modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/translation_manager/TranslationManager.tsx:76]: that's true!

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/translation_manager/TranslationManager.tsx [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+		});
+	}, [selectedLanguageId]);
+
+	return (
+		<TranslationAdminSelector
+			activeLanguageIds={locales.map(({id}) => id)}
+			availableLocales={locales}
+			defaultLanguageId={defaultLanguageId}
+			horizontalDisplay
+			onSelectedLanguageIdChange={(id) => setSelectedLanguageId(id)}
```
[the remark this answers is not in the document]
Comment (veroglez) [reply modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/translation_manager/TranslationManager.tsx:95]: yess! that's true!

File: modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/translation_manager/TranslationAdminSelector.tsx [hunk 8]
```diff
@@ -6,20 +6,30 @@
...
 import {Locale, Translations} from './TranslationAdminContent';
 import TranslationAdminModal from './TranslationAdminModal';
+import TranslationAdminStatusLabel from './TranslationAdminStatusLabel';
 
 interface IProps extends Translations {
 	adminMode?: boolean;
-	onActiveLanguageIdsChange?: (languageIds: string[]) => void;
-	onSelectedLanguageIdChange?: (languageId: string) => void;
-	selectedLanguageId: string;
+	horizontalDisplay?: boolean;
```
[the remark this answers is not in the document]
Comment (p2kmgcl) [reply modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/translation_manager/TranslationAdminSelector.tsx:19]: @veroglez :smiling_face_with_tear:

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/css/main.scss [hunk 9]
```diff
@@ -124,6 +124,10 @@ $productMenuWidth: 320px;
 				border: 0;
 			}
 
+			.translation-manager {
+				width: 115px;
```
[the remark this answers is not in the document]
Comment (p2kmgcl) [reply modules/apps/journal/journal-web/src/main/resources/META-INF/resources/css/main.scss:128]: @veroglez :smiling_face_with_tear:

File: modules/apps/layout/layout-js-components-web/src/main/resources/META-INF/resources/js/components/translation_selector/StatusLabel.tsx [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+	defaultLanguageId,
+	item,
+	translations,
+}: Props) {
+	let status: Status = LABEL_STATUS.notTranslated;
+
+	if (item.id === defaultLanguageId) {
+		status = LABEL_STATUS.default;
+	}
+	else if (item.translations) {
```
[the remark this answers is not in the document]
Comment (veroglez) [reply modules/apps/layout/layout-js-components-web/src/main/resources/META-INF/resources/js/components/translation_selector/StatusLabel.tsx]: This component is also removed

**Comment by veroglez** [comment]
Sent here https://github.com/liferay-echo/liferay-portal/pull/14738

**Pull request closed without merging**